centreon/src/App/Shared/Domain/Exception/AggregateDoesNotExistException.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * Copyright 2005 - 2025 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace App\Shared\Domain\Exception;
+
+abstract class AggregateDoesNotExistException extends \RuntimeException
+{
+    /**
+     * @param array<string, mixed> $criteria
+     */
+    public function __construct(
+        public readonly array $criteria,
+        string $message = 'Resource does not exists.',
+        int $code = 0,
+        ?\Throwable $previous = null,
+    ) {
+        parent::__construct($message, $code, $previous);
+    }
```

Comment:
code is useless, I suggest : 
```suggestion
    public function __construct(
        public readonly array $criteria,
        string $message = 'Resource does not exists.',
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, previous: $previous);
    }
```